src/IO/H5/File.hpp

```diff
@@ -282,7 +281,12 @@ template <AccessType Access_t>
 template <typename ObjectType>
 std::tuple<bool, detail::OpenGroup, std::string>
 H5File<Access_t>::check_if_object_exists(const std::string& path) const {
-  std::string name_only = file_system::get_file_name(path);
+  std::string name_only;
+  if (path == "/") {
```

Comment:
Could you factor this into a separate commit and make sure there's a test for it?